hub/core/chunk_engine.py

```diff
@@ -471,27 +485,75 @@ def update(
         global_sample_indices = tuple(index.values[0].indices(self.num_samples))
         for i, sample in enumerate(samples):
             global_sample_index = global_sample_indices[i]  # TODO!
-            chunks = self.get_chunks_for_sample(global_sample_index, copy=True)
-            if len(chunks) > 1:
-                raise NotImplementedError(
-                    "You can't update a sample that is present in multiple chunks."
+            chunk_ids = enc[global_sample_index]
```

Comment:
can we refactor some of this code out to separate methods? This method has exploded in size and is tough to follow